common/tools/eslint-plugin-azure-sdk/src/rules/ts-package-json-files-required.ts

```diff
@@ -46,37 +72,54 @@ export = {
 
             const nodeValue = node.value;
             const elements = nodeValue.elements as Literal[];
-            const elementValues = elements.map((element: Literal): unknown => element.value);
+            let elementValues = elements.map((element: Literal): unknown => element.value);
 
-            // looks for 'dist' with optional leading './' and optional trailing '/'
-            if (
-              elements.every(
-                (element: Literal): boolean =>
-                  !/^(.\/)?((dist\/)|(dist$))/.test(element.value as string)
-              )
-            ) {
-              context.report({
-                node: nodeValue,
-                message: "dist is not included in files",
-                fix: (fixer: Rule.RuleFixer): Rule.Fix => {
-                  elementValues.push("dist");
-                  return fixer.replaceText(nodeValue, arrayToString(elementValues));
+            let currBadPatterns: string[] = [];
+            let currRequiredPatterns = [...requiredPatterns];
+            const fullMatchIndex = 0;
+            const patternRootMatchIndex = 1;
+            elements.forEach((element) => {
+              const patternMatchResult = (element.value as string).match(regExprStr);
+              if (patternMatchResult !== null) {
+                const patternRoot = patternMatchResult[patternRootMatchIndex];
+                if (badPatterns.indexOf(patternRoot) >= 0) {
```

Comment:
Could you leave some inline comments here to document this logic. It's difficult for me to follow.

Reply:
I feel like some of this could be decomposed into helper functions

Reply:
I did some refactoring and added comments around the file, let me know if it is better to follow now.